src/miner.cpp

```diff
@@ -404,108 +399,26 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const MinerAddre
-                    {
-                        LogPrintf("INFO: missing coins for %s", txin.prevout.hash.GetHex());
-                        // This should never happen; all transactions in the memory
-                        // pool should connect to either transactions in the chain
-                        // or other transactions in the memory pool.
-                        if (!mempool.mapTx.count(txin.prevout.hash))
-                        {
-                            LogPrintf("ERROR: mempool transaction missing input\n");
-                            if (fDebug) assert("mempool transaction missing input" == 0);
-                            fMissingInputs = true;
-                            if (porphan)
-                                vOrphan.pop_back();
-                            break;
-                        }
-
-                        // Has to wait for dependencies
-                        if (!porphan)
-                        {
-                            // Use list for automatic deletion
-                            vOrphan.push_back(COrphan(&tx));
-                            porphan = &vOrphan.back();
-                        }
-                        mapDependers[txin.prevout.hash].push_back(porphan);
-                        porphan->setDependsOn.insert(txin.prevout.hash);
-                        nTotalIn += mempool.mapTx.find(txin.prevout.hash)->GetTx().vout[txin.prevout.n].nValue;
-                        continue;
-                    }
-                    const CCoins* coins = view.AccessCoins(txin.prevout.hash);
-                    assert(coins);
-
-                    CAmount nValueIn = coins->vout[txin.prevout.n].nValue;
-                    nTotalIn += nValueIn;
-
-                    int nConf = nHeight - coins->nHeight;
-
-                    dPriority += (double)nValueIn * nConf;
-                }
-                nTotalIn += tx.GetShieldedValueIn();
-
-                if (fMissingInputs) continue;
-
-                // Priority is sum(valuein * age) / modified_txsize
-                unsigned int nTxSize = ::GetSerializeSize(tx, SER_NETWORK, PROTOCOL_VERSION);
-                dPriority = tx.ComputePriority(dPriority, nTxSize);
-
-                uint256 hash = tx.GetHash();
-                mempool.ApplyDeltas(hash, dPriority, nTotalIn);
-
-                CAmount feePaid = nTotalIn - tx.GetValueOut();
-                CFeeRate feeRate(feePaid, nTxSize);
-
-                if (porphan)
-                {
-                    porphan->dPriority = dPriority;
-                    porphan->feeRate = feeRate;
-                    porphan->feePaid = feePaid;
-                }
-                else
-                    vecPriority.push_back(TxPriority(dPriority, feeRate, feePaid, &(mi->GetTx())));
+                double dPriority = mi->GetPriority(nHeight);
+                CAmount dummy;
+                mempool.ApplyDeltas(mi->GetTx().GetHash(), dPriority, dummy);
```

Comment:
This is undefined behaviour I think (because dummy is uninitialized and `CAmount` is a typedef to a non-class type).